### PR TITLE
Config toggles

### LINKED ISSUE #112: Config options for the toggles in the upper bar
As a user I want to configure my preferences for the toggles in the upper bar:

Follow: ON/OFF
Highlights: ON/OFF
SID: 6581/8580
Timing: PAL/NTSC

Maybe less useful, but include for completeness?
~~Mode: FLAT/SHARP~~
~~Octave~~

### PR DESCRIPTION
**Pull request by micheldebree**
Added four config option:

`Editor.Follow.Play` to set default follow play on/off.
`Editor.Sequence.Highlights` to set default sequence highlights on/off.
`Sound.Emulation.Default.Model` to set the default SID model (6581/8580) on startup.
`Sound.Emulation.Default.Region` to set default region (PAL/NTSC) on startup.

This has also got the `config_singleton` branch merged into it.

Resolves #112

### CODE REVIEW

File: SIDFactoryII/source/utils/configfile.h [hunk 1]
```diff
@@ -19,6 +19,10 @@ namespace Utility
 	public:
 		ConfigFile(const Foundation::IPlatform& inPlatform, const std::string& inFilename, const std::vector<std::string>& inValidSectionTags);
 
+		ConfigFile(ConfigFile& inOther) = delete;
+		ConfigFile(ConfigFile&& inOther) = delete;
+		ConfigFile(const ConfigFile& inOther) = delete;
+
```
Comment (TheRealLaxity) [inline SIDFactoryII/source/utils/configfile.h:25]: Probably also need to delete the assingment and move operators. 

The non const copy comstructor you do not need to delete, as it's not a default constructor.

File: SIDFactoryII/source/utils/global.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+{
+	// Singleton for global configuration
+	class Global
+	{
+
+	public:
+		static Global& instance();
+		Global(Global& inOther) = delete;
+		Global(Global&& inOther) = delete;
+		Global(const Global& inOther) = delete;
```
Comment (TheRealLaxity) [inline SIDFactoryII/source/utils/global.h:22]: Same as the other constructors with the same pattern.

**Review by TheRealLaxity — APPROVED** [review]
This looks good, except for the copy/move guard. We can post-fix those.

File: SIDFactoryII/source/utils/configfile.h (hunk 1 above)
Comment (micheldebree) [reply SIDFactoryII/source/utils/configfile.h:25]: Deleted https://en.cppreference.com/w/cpp/language/move_assignment operator
Removed the line `ConfigFile(const ConfigFile& inOther) = delete;`

File: SIDFactoryII/source/utils/global.h (hunk 2 above)
Comment (micheldebree) [reply SIDFactoryII/source/utils/global.h]: Deleted https://en.cppreference.com/w/cpp/language/move_assignment operator
Removed the line `Global(const Global& inOther) = delete;`

**Pull request merged**